File: go/config/Voice.go
package config

import (
	"github.com/lvcdy/huawei-lte-api-go/session"
)

// Voice 对应 config/Voice.py。
type Voice struct {
	*session.ApiGroup
}

// NewVoice 创建 Voice API 分组。
func NewVoice(s *session.Session) *Voice {
	return &Voice{ApiGroup: session.NewApiGroup(s)}
}

// Config 对应 voice/config.xml。
func (v *Voice) Config() (map[string]interface{}, error) {
	return v.S.Get("voice/config.xml", nil, "config")
}

// Country 对应 voice/country.xml。
func (v *Voice) Country() (map[string]interface{}, error) {
	return v.S.Get("voice/country.xml", nil, "config")
}

File: src/config/voice.rs
//! Voice config group (`config/Voice.py`).

use serde_json::Value;

use crate::api::get_value;
use crate::connection::Connection;
use crate::errors::Result;

/// Voice config group.
pub struct VoiceConfig<'a> {
    conn: &'a Connection,
}

impl<'a> VoiceConfig<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        VoiceConfig { conn }
    }

    /// `voice/config.xml`. Voice configuration.
    pub fn config(&self) -> Result<Value> {
        get_value(self.conn, "config/voice/config.xml")
    }

    /// `voice/country.xml`. Voice country.
    pub fn country(&self) -> Result<Value> {
        get_value(self.conn, "config/voice/country.xml")
    }
}
